### PR TITLE
Added CLI run functionality for apsim.dll

### LINKED ISSUE #6987: Missing CLI run functionality for apsim.dll
Run method for apsim.dll CLI isn't implemented.

### PR DESCRIPTION
**Pull request by dexy58**
Fixes #6987

### CODE REVIEW

**Comment by dexy58** [comment]
@hol430 Hello, can you please check why TestSendDataTable test fails? I can't seem to figure it out (I haven't even touched any of the methods this Test calls.

**Comment by hol430** [comment]
Maybe a race condition in the test...

retest this please jenkins

**Comment by dexy58** [comment]
@hol430 Thank you :)
All tests have passed now.

File: APSIM.Cli/Program.cs [hunk 1]
```diff
@@ -53,7 +54,31 @@ private static void HandleParseError(IEnumerable<Error> errors)
...
+            {
+                files = options.Files;
+            }
+            foreach (string file in files)
+            {
+                Simulations sims = FileFormat.ReadFromFile<Simulations>(file, e => throw e, false);
+                if (Path.GetExtension(file) == ".json")
+                {
+                    sims.Links.Resolve(sims, true, true, false);
+                }
```
Comment (hol430) [inline APSIM.Cli/Program.cs:72]: I don't think we need to resolve links for .json files here. I think we only do that in the `Document()` function as a bit of a hack, but it shouldn't be necessary for running files - only for documenting them.

```suggestion
```

**Review by hol430 — CHANGES_REQUESTED** [review]
Thanks for the contribution, @dexy58!

File: APSIM.Cli/Program.cs (hunk 1 above)
Comment (dexy58) [reply APSIM.Cli/Program.cs]: Commited this change. Tested it and it works without this part.

**Review by hol430 — APPROVED** [review]

**Pull request merged**